src/ObjectPool/src/DefaultObjectPoolProvider.cs

```diff
@@ -11,6 +11,11 @@ public class DefaultObjectPoolProvider : ObjectPoolProvider
 
         public override ObjectPool<T> Create<T>(IPooledObjectPolicy<T> policy)
         {
+            if (typeof(IDisposable).IsAssignableFrom(typeof(T)))
```

Comment:
This should also have argument validation for `policy`. I know it will just throw later, but for maintenance reasons we usually like to validate arguments at public entry points.

Reply:
Done -- will be updated with the next push.